bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/ChannelTypeConverter.java

```diff
@@ -107,8 +109,12 @@ private StateDescription readStateDescription(NodeIterator nodeIterator) {
     protected ChannelTypeXmlResult unmarshalType(HierarchicalStreamReader reader, UnmarshallingContext context,
             Map<String, String> attributes, NodeIterator nodeIterator) throws ConversionException {
 
-        ChannelTypeUID channelTypeUID = new ChannelTypeUID(super.getUID(attributes, context));
-        boolean advanced = isAdvanced(attributes, false);
+        boolean advanced = readBoolean(attributes, "advanced", false);
+        boolean system = readBoolean(attributes, "system", false);
+
+        String uid = system ? String.format("%s:%s", SystemChannelTypeProvider.NAMESPACE, super.getID(attributes))
```

Comment:
Could you extract the creation of the system channel type UID into a separate method please: so that this code looks like this: `String uid = system ? getSystemChannelTypeId(super.getID(attributes)) : super.getID(attributes))`